### ISSUE TITLE
Trying to implement new features, but I'm having trouble with the initial connection

### ISSUE THREAD

**Issue by drewbeller** [groundmelon/pydjisdk#1]
It throws the error of not receiving an ACK signal back from the drone (I'm using a Matrice 100 that's already been activated via the QT sample and DJI-Go app). I'm not entirely sure that the drone is getting any signal at all to begin with. Has anyone else seen this issue?

Could it be that I need to replace the bundle_id (line 104 in SDKApplication.py) with my App_key? What purpose does the bundle_id serve?

Here is the feedback I receive in the logs after issuing the get_api_version command (which works for me in the Linux commandline example):

```
2016-06-15 13:14:02,540 [     app] DEBUG Init CRC with poly=0x18005 initval=0x3AA3
2016-06-15 13:14:02,540 [     app] DEBUG Init CRC with poly=0x104C11DB7 initval=0x3AA3
2016-06-15 13:14:02,562 [     app] DEBUG Thread-2(<class 'pydjisdk.SerialPort.SerialListener'>) will start.
2016-06-15 13:14:02,562 [app.parser] DEBUG Thread-3(<class 'pydjisdk.Protocol.ProtocolParser'>) will start
2016-06-15 13:14:05,316 [ app.ss2] DEBUG Send [AA130002000000000000007E00000151E99C05]
2016-06-15 13:14:05,316 [ app.ss2] DEBUG sof:170, length:19, ver:0, session:2, ack:0, res0:0, pad:0, enc:0, res1:0, seq:0, crc16:32256
2016-06-15 13:14:05,316 [ app.ss2] DEBUG <SessionThread> session_id[2] #0 will start
2016-06-15 13:14:06,317 [ app.ss2] DEBUG Send [AA130002000000000000007E00000151E99C05]
2016-06-15 13:14:06,317 [ app.ss2] DEBUG sof:170, length:19, ver:0, session:2, ack:0, res0:0, pad:0, enc:0, res1:0, seq:0, crc16:32256
2016-06-15 13:14:07,318 [ app.ss2] DEBUG Send [AA130002000000000000007E00000151E99C05]
2016-06-15 13:14:07,319 [ app.ss2] DEBUG sof:170, length:19, ver:0, session:2, ack:0, res0:0, pad:0, enc:0, res1:0, seq:0, crc16:32256
2016-06-15 13:14:08,319 [ app.ss2] DEBUG Send [AA130002000000000000007E00000151E99C05]
2016-06-15 13:14:08,320 [ app.ss2] DEBUG sof:170, length:19, ver:0, session:2, ack:0, res0:0, pad:0, enc:0, res1:0, seq:0, crc16:32256
2016-06-15 13:14:08,320 [     app] WARNING Get api version failed.
2016-06-15 13:14:08,320 [ app.ss2] WARNING Receive ack failed. <SessionThread> session_id[2] #0
2016-06-15 13:14:08,320 [ app.ss2] DEBUG <SessionThread> session_id[2] #0 will stop.
```

I also get this error thrown occasionally when running app.py:

```
Choose operation:Exception in thread Thread-3:
Traceback (most recent call last):
  File "/usr/lib/python2.7/threading.py", line 810, in __bootstrap_inner
    self.run()
  File "/home/parallels/Documents/pydjisdk/pydjisdk/Protocol.py", line 294, in run
    self.feed(buf)
  File "/home/parallels/Documents/pydjisdk/pydjisdk/Protocol.py", line 239, in feed
    assert(len(self.buf) < 1024)
AssertionError
```

**Comment by groundmelon** [comment]
Could you please verify that you've filled in the right parameters in `config.yaml` ? Especially the path for your serial port?

**Comment by groundmelon** [comment]
Could you please verify that you've filled in the right parameters in config.yaml ? Especially the path for your serial port?

Also, make sure that your baudrate is compatible with your drone.

The second assertion happens when it cannot find enough data in correct format for parsing even when there is 1024 bytes in the receive buffer. Maybe the income data from the serial port is meaningless or not from a M100 or not in correct baudrate.

The newest commit allows you to modify baudrate in `config.yaml`

**Comment by drewbeller** [comment]
Turns out the error was caused by a low battery! For some reason it continued to work with the controller. It seems that when the battery drops to the last bar, the m100 will lock access to the serial port. After I charged the battery everything seems to be working. 

Changing the baud rate to 115200 on both the drone and in the app seems to have gotten rid of the second error. I found a post from DJI saying that the only fully supported baud over UART is 115200, before I was operating on 230400 (which is set as the default on the m100).
